src/Controller/Api/ApiController.php

```diff
@@ -155,11 +131,35 @@ private function formatUri(string $product): string
         foreach ($productIdSubStr as $symbol) {
             if ((int)$symbol >= 1) {
                 $productId[] = $symbol;
-            } elseif (is_int($symbol) && !empty($productId)) {
+            } elseif (!empty($productId)) {
                 $productId[] = $symbol;
             }
         }
 
         return $productName . implode($productId);
     }
+    
+    /**      
+     * Translates keys of the Product object.
+     * 
+     * @param array $translateIdentifiers
+     * @param $translator  
+     * 
+     * @return array
+     */
+    private function translate(array $translateIdentifiers, $translator) : array
```

Comment:
The method is redundant
- the translator interface is not specified
- it's not clear what should be transferred to translateIdentifiers
- and the presence of this method forces you to iterate over an array two times, at first to get Ids, then to translate them.